File: src/sum/Sum.java
package sum;

public class Sum {
    private int total = 0;

    public Sum(int n) {
        this.total = n;
    }

    public Sum sum(int n) {
        total += n;
        return this;
    }

    public int calculate() {
        return total;
    }

    public static void main(String[] args) {
        int result = new Sum(1).sum(34).sum(5).sum(108).calculate();
        System.out.println(result); // Output: 148
    }
}

File: src/sum/sum.ts
// Function sum, which takes an optional parameter n of type number
function sum(n?: number) {
    // Initialize total with the provided number n, or 0 if not provided
    let total = n || 0;

    // Define a nested function addNext, which takes an optional parameter num of type number
    function addNext(num?: number) {
        // If num is not provided, return the current total sum
        if (num === undefined) {
            return total;
        }
        // If num is provided, add it to the total sum
        total += num;
        // Return the addNext function itself to allow chaining
        return addNext;
    }

    // Return the addNext function to allow initial chaining
    return addNext;
}

// Example usage:
const result = sum(1)(34)(5)(108)();
console.log(result); // Output: 148
